packages/@ember/array/index.ts

```diff
@@ -1193,6 +1193,25 @@ interface EmberArray<T> extends Enumerable {
   */
   without(value: T): NativeArray<T>;
 }
+
+const emberArrayBrackets = nonEnumerableComputed({
+  get() {
+    return this;
+  },
+  set(_key, value) {
+    this.replace(0, this.length, value);
+    return this;
+  },
+});
+
+const emberArrayFirstObject = nonEnumerableComputed(function () {
+  return objectAt(this, 0);
+}).readOnly();
+
+const emberArrayLastObject = nonEnumerableComputed(function () {
+  return objectAt(this, this.length - 1);
+}).readOnly();
```

Comment:
I don't love how we're assigning them to variables here, but I can't find any other good way to get a reference to their CPs.